### PR TITLE
Implement pre-approval flow in btc-staker

### PR DESCRIPTION
**Pull request by KonradStaniec**
Implements pre-approval flow in btc staker.

There is new config option `CheckActiveInterval` (cc: @filippos47 )

In general this is basic support which assumes there is vigiliante in the network that wil provide the proof on inclusion at some point in time.

At this time two flows are supported, to use pre-approval flow staker must provide flag:
`--send-to-babylon-first=true`

At least two follow up tasks (which is not blocking for devnet2):
- https://github.com/babylonlabs-io/btc-staker/issues/43 it turns out we always send staking transaction with witness to Babylon which could inflate inital delegation cost
- https://github.com/babylonlabs-io/btc-staker/issues/45

### CODE REVIEW

File: itest/e2e_test.go [hunk 1]
```diff
@@ -1451,7 +1535,7 @@ func TestStakingUnbonding(t *testing.T) {
 	tm.waitForStakingTxState(t, txHash, proto.TransactionState_SPENT_ON_BTC)
 }
 
-func TestUnbondingRestartWaitingForSignatures(t *testing.T) {
+func AestUnbondingRestartWaitingForSignatures(t *testing.T) {
```
Comment (Lazar955) [inline itest/e2e_test.go:1538]:
```suggestion
func TestUnbondingRestartWaitingForSignatures(t *testing.T) {
```

File: staker/babylontypes.go [hunk 2]
```diff
@@ -278,3 +281,188 @@ func (app *StakerApp) finalityProviderExists(fpPk *btcec.PublicKey) error {
...
+
+	return signed, nil
+}
+
+// activateVerifiedDelegation must be run in separate goroutine whenever delegation
+// reaches verified state. i.e
+// - delegation is on babylon
+// - delegation has received enough covenant signatures
+func (app *StakerApp) activateVerifiedDelegation(
+	stakerAddress btcutil.Address,
```
Comment (Lazar955) [inline staker/babylontypes.go:307]: stakerAddress seems to be unused

**Review by Lazar955 — COMMENTED** [review]
lgtm! Nice work

**Review by Lazar955 — APPROVED** [review]

File: staker/babylontypes.go [hunk 3]
```diff
@@ -133,7 +138,7 @@ func (app *StakerApp) buildDelegation(
 	stakerAddress btcutil.Address,
 	storedTx *stakerdb.StoredTransaction) (*cl.DelegationData, error) {
 
-	stakingTxInclusionProof := app.mustBuildInclusionProof(req)
+	// stakingTxInclusionProof := app.mustBuildInclusionProof(req)
```
Comment (RafilxTenfen) [inline staker/babylontypes.go:141]: leftover?

**Review by RafilxTenfen — APPROVED** [review]
:rocket:

**Pull request merged**